### PR TITLE
CloudFormation Resource Specification 14.2.0

### PR DESCRIPTION
**Pull request by kddejong**
*Issue #, if available:*

*Description of changes:*
- Update specs to 14.2.0

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by kddejong** [comment]
`Specified property type AWS::SSM::Association.ParameterValues not found for property Parameters`

**Comment by PatMyron** [comment]
> `Specified property type AWS::SSM::Association.ParameterValues not found for property Parameters`

@kddejong [`AWS::SSM::Association.Parameters`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ssm-association.html#cfn-ssm-association-parameters) is definitely malformed in [resource specification](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html) 14.2.0

[I switched `AWS::SSM::Association.Parameters` from `"ItemType": "ParameterValues"` to `"PrimitiveType": "String"` for the Visual Studio Code extension](https://github.com/aws-cloudformation/aws-cfn-lint-visual-studio-code/pull/101)

File: src/cfnlint/data/ExtendedSpecs/all/01_spec_patch.json [hunk 1]
```diff
@@ -119,5 +119,22 @@
     "op": "replace",
     "path": "/ResourceTypes/AWS::ResourceGroups::Group/Properties/Tags/ItemType",
     "value": "Tag"
+  },
+  {
+    "op": "add",
+    "path": "/PropertyTypes/AWS::SSM::Association.ParameterValues",
```
Comment (kddejong) [inline src/cfnlint/data/ExtendedSpecs/all/01_spec_patch.json:125]: @PatMyron this is what it was previously.  It used to be in the spec but got dropped for some reason in this release.

Comment (PatMyron) [reply src/cfnlint/data/ExtendedSpecs/all/01_spec_patch.json:125]: copying the other snippet for reference:
```json
        "Parameters": {
          "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ssm-association.html#cfn-ssm-association-parameters",
          "UpdateType": "Mutable",
          "Required": false,
          "Type": "Map",
          "ItemType": "ParameterValues"
        },
```

**Pull request merged**